src/tools/sfdk/script.cpp

```diff
@@ -260,6 +262,32 @@ class JSBuildEngineExtension : public QObject
     }
 };
 
+class JSSignExtension : public QObject
+{
+    Q_OBJECT
+public:
+    using QObject::QObject;
+    Q_INVOKABLE bool exportKey() const
+    {
+        if (!SdkManager::hasEngine()) {
+            qCWarning(sfdk).noquote() << SdkManager::noEngineFoundMessage();
+            return false;
+        }
+
+        QString keyName;
+        QString keyPassFile;
+        // TODO: HACK ??
+        for (const auto &option : Configuration::effectiveState()) {
```

Comment:
@martyone what do you think about exporting key here and getting the arguments?

Reply:
I think we should instead add

```
void BuildEngine::importGpgKey(const QString &id, const Utils::FilePath &passphraseFile, const QObject *context,
            const Functor<bool> &functor);
```

expose it on `JSBuildEngineExtension` with help of `execAsynchronous` and implement the rest in javascript:

```javascript
export function maybeImportSigningKey() {
    if (!configuration.isOptionSet("package.signing-user"))
        return [true, ""];

    var signingUser = configuration.optionArgument("package.signing-user");
    var signingPassphraseFile = configuration.isOptionSet("package.signing-passphrase-file")
        ? configuration.optionArgument("package.signing-passphrase-file")
        : "";
    if (!engine.importGpgKey(signingUser, signingPassphraseFile))
        return [false, qsTr("Failed to share GnuPG key with the build engine")];

    return [true, ""];
}
```

We should also think about being more verbose about the reasons for a failure - gpg may print something useful on stderr.